### PR TITLE
created Todo class

### PR DESCRIPTION
**Pull request by vdimition**

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -0,0 +1,10 @@
+<!DOCTYPE html>
+<html lang="en">
+<head>
+  <meta charset="UTF-8">
+  <title>TODO</title>
+</head>
+<body>
+<script src="./main.js"></script>
+</body>
+</html>
```
Comment (gestap) [inline index.html]: the document must end with one empty line

File: main.js [hunk 2]
```diff
@@ -0,0 +1,51 @@
+class Todo {
```
Comment (ost-ptk) [inline main.js]: use strict mode

File: main.js [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+    this.todos.push({"id": this.id, "title": title, "priority": priority});
+    this.id++;
+  }
+
+  removeItem(id) {
+    const index = this.todos.findIndex(item => item.id === id);
+    if (index >= 0) {
+      this.todos.splice(index, 1)
+      return true;
+    } else {
```
Comment (gestap) [inline main.js]: no need for this else

File: main.js [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+      return false;
+    }
+  }
+
+  getItem(id) {
+    const index = this.todos.findIndex(item => item.id === id);
+    if (index >= 0) {
+      return this.todos[index];
+    } else {
```
Comment (gestap) [inline main.js]: no need for this else

File: main.js [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+      return null;
+    }
+  }
+
+  next() {
+    if (this.todos.length) {
+      let maxPriority = {"priority": -Infinity};
+      this.todos.map(item => {
```
Comment (gestap) [inline main.js]: suggestion: you can use `sort` method

File: main.js [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+        if (item.priority > maxPriority.priority) {
+          maxPriority = item;
+        }
+      });
+      return maxPriority;
+    } else {
+      return console.log('There are no items in the list');
```
Comment (gestap) [inline main.js]: you need to `...throw an error` instead return undefined

**Review by ost-ptk — CHANGES_REQUESTED** [review]

File: main.js [hunk 7]
```diff
@@ -0,0 +1,43 @@
+'use strict';
+
+class Todo {
+  constructor() {
+    this.todos = [];
+    this.id = 1;
+  }
+
+  addItem(title, priority) {
```
Comment (ost-ptk) [inline main.js:9]: `the method returns the unique id assigned to the item (positive integer);`

File: main.js [hunk 8]
```diff
@@ -0,0 +1,43 @@
...
+    this.todos = [];
+    this.id = 1;
+  }
+
+  addItem(title, priority) {
+    this.todos.push({"id": this.id, "title": title, "priority": priority});
+    this.id++;
+  }
+
+  removeItem(id) {
```
Comment (ost-ptk) [inline main.js]: `it should return ... or false if there was no such item`

File: main.js [hunk 9]
```diff
@@ -0,0 +1,43 @@
...
+    const index = this.todos.findIndex(item => item.id === id);
+    if (index >= 0) {
+      this.todos.splice(index, 1);
+      return true;
+    }
+  }
+
+  getItem(id) {
```
Comment (ost-ptk) [inline main.js]: `if there is no such item, it should return null`

File: main.js [hunk 10]
```diff
@@ -0,0 +1,43 @@
...
+      return this.todos[index];
+    }
+  }
+
+  next() {
+    if (this.todos.length) {
+      const max = Math.max(this.todos);
+      return max;
+    }
+    return console.log('There are no items in the list');
```
Comment (gestap) [inline main.js]: `if there are no items in the list, throw an error`
you return `undefined`

File: main.js [hunk 11]
```diff
@@ -0,0 +1,56 @@
...
+
+  next() {
+    try {
+      const max = this.todos.sort((a,b) => b["priority"] - a["priority"]);
+      if (!this.todos.length) {
+        throw new SyntaxError('There are no items in the list');
+      }
+      return max[0];
+    } catch (e) {
+      return console.log(e.message);
```
Comment (gestap) [inline main.js]: you return `undefined` instead of `throw error`

Comment (gestap) [reply main.js]: `console.log(e.message)` - return `undefined`

File: main.js [hunk 12]
```diff
@@ -0,0 +1,56 @@
...
+  getItem(id) {
+    const index = this.todos.findIndex(item => item.id === id);
+    if (index >= 0) {
+      return this.todos[index];
+    }
+    return null;
+  }
+
+  next() {
+    try {
```
Comment (gestap) [inline main.js]:
```suggestion
```

File: main.js [hunk 13]
```diff
@@ -0,0 +1,56 @@
...
+      const max = this.todos.sort((a,b) => b["priority"] - a["priority"]);
+      if (!this.todos.length) {
+        throw new SyntaxError('There are no items in the list');
+      }
+      return max[0];
+    } catch (e) {
```
Comment (gestap) [inline main.js]:
```suggestion
```

File: main.js [hunk 14]
```diff
@@ -0,0 +1,56 @@
...
+      console.log(e.message);
```
Comment (gestap) [inline main.js]:
```suggestion
```

File: main.js [hunk 15]
```diff
@@ -0,0 +1,56 @@
...
+    }
```
Comment (ost-ptk) [inline main.js:37]:
```suggestion
```

**Review by ost-ptk — APPROVED** [review]

**Pull request closed without merging**